clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5087,6 +5087,27 @@ static void handleSYCLRegisterNumAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
   D->addAttr(::new (S.Context) SYCLRegisterNumAttr(S.Context, AL, RegNo));
 }
 
+static void handleSYCLIntelESimdWidenAttrAttr(Sema &S, Decl *D,
```

Comment:
Can you please rewrite this to use 'Add' and 'Merge' methods. You can see https://github.com/intel/llvm/pull/3224 for examples. This aligns with community standards better and will help us with our upstreaming efforts.

Also, I don't think redeclarations are currently handled in this code i.e. cases like - 
```
[[intel::sycl_esimd_widen(8)]] void func();
[[intel::sycl_esimd_widen(16)]] void func() {}
```
You can add a 'Merge' method for this.

Reply:
Done